aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePluginConfiguration.java

```diff
@@ -18,53 +18,134 @@
         this.syncMode = syncMode;
+        this.baseSyncIntervalMs = baseSyncIntervalMs;
     }
 
     static AWSDataStorePluginConfiguration fromJson(JSONObject pluginJson) throws DataStoreException {
-        // If no sync mode is specified, we just use the default (no sync) and continue
-        if (pluginJson == null || !pluginJson.has("syncMode")) {
-            return new AWSDataStorePluginConfiguration(SyncMode.LOCAL_ONLY);
-        }
+        SyncMode syncMode = DEFAULT_SYNC_MODE;
+        long baseSyncIntervalMs = DEFAULT_BASE_SYNC_INTERVAL_MS;
 
-        try {
-            // If user has specified a sync mode, find out what it was.
-            final SyncMode syncMode = SyncMode.fromJsonPropertyValue(pluginJson.getString("syncMode"));
+        if (pluginJson == null) {
+            return new AWSDataStorePluginConfiguration(syncMode, baseSyncIntervalMs);
+        }
 
-            return new AWSDataStorePluginConfiguration(syncMode);
-        } catch (JSONException exception) {
-            throw new DataStoreException(
+        final Iterator<String> jsonKeys = pluginJson.keys();
+        while (jsonKeys.hasNext()) {
+            final String keyString = jsonKeys.next();
+            final ConfigKey configKey;
+            try {
+                configKey = ConfigKey.fromString(keyString);
+            } catch (IllegalArgumentException noSuchConfigKey) {
+                throw new DataStoreException(
+                    "Saw unexpected config key: " + keyString,
+                    "Make sure your amplifyconfiguration.json is valid."
+                );
+            }
+            try {
+                switch (configKey) {
+                    case SYNC_MODE:
+                        String jsonValue = pluginJson.getString(ConfigKey.SYNC_MODE.toString());
+                        syncMode = SyncMode.fromJsonPropertyValue(jsonValue);
+                        break;
+                    case BASE_SYNC_INTERVAL_MS:
+                        baseSyncIntervalMs = pluginJson.getLong(ConfigKey.BASE_SYNC_INTERVAL_MS.toString());
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Unsupported config key = " + configKey.toString());
+                }
+            } catch (JSONException jsonException) {
+                throw new DataStoreException(
                     "Issue encountered while parsing configuration JSON",
-                    exception,
-                    "Check the attached exception for more details."
-            );
+                    jsonException, "Ensure your amplifyconfiguration.json is valid."
+                );
+            }
         }
 
+        return new AWSDataStorePluginConfiguration(syncMode, baseSyncIntervalMs);
     }
 
     /**
      * Gets the synchronization mode.
      * @return Synchronization mode
      */
     @NonNull
-    SyncMode getSyncMode() {
+    SyncMode syncMode() {
```

Comment:
Is there a reason you're dropping the get verb prefix? I debated this on my current Auth work and landed on keeping the get at the beginning of the method name since, in addition to being consistent with our pattern of starting method names with verbs, I also think it's a better experience for the developer if all getter methods start with get and thus are grouped together in autosuggest.

Reply:
Thanks for bringing this up.

Currently, we have inconsistent naming across the codebase for accessor methods. Some use the "`OkHttp` style" `foo()` (no `get`), and some use the style of prefixing by `getFoo()`.

Our builders mostly use the `.someValue(...)` configurator style, instead of the `.withSomeValue(...)`, but this is probably another area where we have some inconsistency.

Aligning the accessor methods to one naming style is going to be a large change in our code base. Let's discuss the preferred style amongst the team. For methods that simply returned a cached field value, I do prefer the version without `get...()`, personally.

Reply:
Resolution: starting to move these things to include `get...()`, as per kotlin interop doc:

https://kotlinlang.org/docs/reference/java-interop.html#getters-and-setters